calculator/main.py

```diff
@@ -37,14 +47,18 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    if not isinstance(operation, dict):
+        raise InvalidOperation('Given operation is invalid.')
+
+    else:
```

Comment:
The `else` here is superfluous, the raise in your if statement will prevent any further code from executing if it's triggered.